### PR TITLE
feat(config): allow disabling TLS on clients

### PR DESCRIPTION
**Pull request by freakyfelt**
## Summary

feat: Adds a hidden config option for passing 'insecure' to the client, allowing for local development
feat: Allow specifying the config file path in environment with `CONTENTFUL_CONFIG_FILE`

### CODE REVIEW

File: lib/utils/contentful-clients.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ async function createManagementClient(params) {
     proxyConfig.proxy = proxy
   }
 
-  return createClient({ ...params, ...proxyConfig, host })
+  return createClient({ ...params, ...proxyConfig, host, insecure })
```
[the remark this answers is not in the document]
Comment (freakyfelt) [reply lib/utils/contentful-clients.js:20]: hmm... seems like integration tests are using insecure. Need to figure out why that is triggering

[the remark this answers is not in the document]
Comment (freakyfelt) [reply lib/utils/contentful-clients.js:20]:
```
circleci@845745ffa8e2:~$ cat .contentfulrc.json | grep -a4 insecure
    "port": 3333,
    "isHttps": false
  },
  "rawProxy": true,
  "insecure": true
}
```

Looks like insecure is getting written out to the `.contentfulrc.json`. Need to clear it after whatever test is writing out the state.

[the remark this answers is not in the document]
Comment (phoebeschmidt) [reply lib/utils/contentful-clients.js:20]: nice seems you figured it out 👍

File: test/integration/cmds/config/add.test.js [hunk 2]
```diff
@@ -32,6 +34,16 @@ test('config add throws error when option mt is empty', done => {
     .end(done)
 })
 
+test('config add allows insecure', (done) => {
+  app()
+    .run('config add --insecure=true')
+    .code(0)
+    .end(() => {
+      const context = readSync('$HOME/.contentfulrc.json')
```
Comment (phoebeschmidt) [inline test/integration/cmds/config/add.test.js:42]: do you mean readfilesync maybe? this is failing because it expects first arg to be an integer

Comment (freakyfelt) [reply test/integration/cmds/config/add.test.js]: bah good catch

**Comment by freakyfelt** [comment]
Added the ability to specify the config file path using environment variable `CONTENTFUL_CONFIG_FILE` (e.g. `CONTENTFUL_CONFIG_FILE=/etc/contentful/contentfulrc.json`

**Pull request merged**